src/System.Memory/src/System/Memory.cs

```diff
@@ -246,7 +247,20 @@ public bool TryGetArray(out ArraySegment<T> arraySegment)
         /// allocates, so should generally be avoided, however it is sometimes
         /// necessary to bridge the gap with APIs written in terms of arrays.
         /// </summary>
-        public T[] ToArray() => Span.ToArray();
+        public T[] ToArray()
+        {
+            if (_index < 0)
+            {
+                Span<T> span = ((OwnedMemory<T>)_arrayOrOwnedMemory).AsSpan().Slice(_index & RemoveOwnedFlagBitMask, _length);
+                return span.ToArray();
+            }
+            else
+            {
+                T[] result = new T[_length];
```

Comment:
Did you consider special-casing _length == 0 to return `Array.Empty<T>()`?

Reply:
I did, but since I would have to do something like the following, I skipped the length check.
```C#
#if !netstandard11
            Array.Empty<T>();
#else
            new T[0];
#endif
```

What is the benefit of using `Array.Empty<T>()` over `T[0]`?

I suppose the extra if check won't matter too much for large arrays (since copy is fairly expensive already).

Reply:
`Array.Empty<T>()` returns a cached array; `new T[0]` allocates a new array.

Reply:
It might be worth creating internal static empty array, instead of doing an ifdef. It's going to be very common.

Reply:
It looks like that already exists?
```C#
SpanHelpers.PerTypeValues<T>.EmptyArray
```